dsl/src/commonMain/kotlin/it/unibo/collektive/field/FieldManipulation.kt

```diff
@@ -1,65 +1,67 @@
 package it.unibo.collektive.field
 
-import it.unibo.collektive.field.Field.Companion.hood
-import it.unibo.collektive.field.Field.Companion.reduce
+import it.unibo.collektive.field.Field.Companion.fold
 
 /**
- * Get the minimum value of a field.
- * @param includingSelf if true the local node is included in the computation.
+ * Get the minimum value of a field, excluding the local value, starting from [base].
+ * To consider the local value, explicitly provide it as [base].
  */
-fun <T : Comparable<T>> Field<T>.min(includingSelf: Boolean = true): T = when (includingSelf) {
-    true -> hood { acc, value -> if (value < acc) value else acc }
-    false -> reduce(includingSelf = false) { acc, value -> if (value < acc) value else acc }
-}
+fun <ID : Any, T : Comparable<T>> Field<ID, T>.min(base: T): T =
+    fold(base) { acc, value -> if (value < acc) value else acc }
 
 /**
- * Get the maximum value of a field.
- * @param includingSelf if true the local node is included in the computation.
+ * Get the minimum value of a field, including the local value.
  */
-fun <T : Comparable<T>> Field<T>.max(includingSelf: Boolean = true): T = when (includingSelf) {
-    true -> hood { acc, value -> if (value > acc) value else acc }
-    false -> reduce(includingSelf = false) { acc, value -> if (value > acc) value else acc }
-}
+fun <ID : Any, T : Comparable<T>> Field<ID, T>.minWithSelf(): T = min(localValue)
```

Comment:
@DanySK we had agreed on providing only `min(base)` since most of the time the `min` on a field is intended self-excluded. The `min` with self can be obtained with `min(base)` where `base` represents the local value of the field.

Reply:
I don't mind having a dedicated method with a different name, though. I feel like at some point we would end up writing it in the library...